src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,88 @@
+package duke.parser;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import duke.command.AllCommand;
+import duke.command.AnyCommand;
+import duke.command.AtCommand;
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.EventCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.ToDoCommand;
+import duke.command.UnmarkCommand;
+import duke.exception.DukeException;
+import duke.exception.EmptyCommandException;
+
+/**
+ * Parser for Duke application.
+ *
+ * @author Farrel Dwireswara Salim
+ */
+public class Parser {
+    private boolean isListening;
+
+    /**
+     * Constructs a new instance of Parser.
+     */
+    public Parser() {
+        this.isListening = true;
+    }
+
+    /**
+     * Parses the given string to a Command object.
+     *
+     * @param text the string to be parsed.
+     * @return the Command which represents the corresponding string input.
+     * @throws DukeException If the string to be parsed is not valid.
+     */
+    public Command parseText(String text) throws DukeException {
+        List<String> commands = Arrays.stream(text.trim().split(" ", 2))
+                .map(String::trim)
+                .collect(Collectors.toList());
+
+        String mainCommand = commands.get(0);
+        String description = commands.size() > 1 ? commands.get(1) : "";
+
+        if (mainCommand.equals("bye")) {
```

Comment:
Love how clean this is! But based on this context, maybe a switch statement here can help make it easier for readers to make sense of the code here.